CenterForOpenScience/ember-osf-web
# [No ticket] Preprint branding rework

-   Ticket: []
-   Feature flag: n/a

## Purpose
- Only rely on `brand` relationship for setting preprint colors

## Summary of Changes
- Use `brand.primaryColor` for branded navbar background color
  - Add styling if the brand's primaryColor does not provide sufficient contrast with white text
 - Add special-case for BioHackrXiv to change navbar color to white (their primary color would be white, but that creates problems for `<Button>` behavior, so their primary is black instead)
- Use `brand.heroBackgroundImage` for hero banner background, as well as advisory board section when @brianpilati implements that page. Currently just a solid color
- Use `brand.navbarLogoImage` for navbar logo if `preprintProvider.assets.square_color_transparent` is not defined
- Update `<Button>` component to detect primary color and use white/black text depending on color contrast


## Screenshot(s)
- Thesis Commons
![image](https://github.com/CenterForOpenScience/ember-osf-web/assets/51409893/7c2a5d5a-131e-42a4-96ef-fb6fc9bb9419)

- AgriXiv
![image](https://github.com/CenterForOpenScience/ember-osf-web/assets/51409893/aa7e263f-7007-4f87-b7d0-d7b475074ddb)


- BioHackrXiv
![image](https://github.com/CenterForOpenScience/ember-osf-web/assets/51409893/ee6f4f41-48bd-4330-92b7-325fbd1571e6)


## Side Effects



## QA Notes
- Please ensure that color contrast ratio is WCAG AA compliant for preprint navbars in the new discover page.
- Please spot-check branded registries and collection providers to make sure that buttons used in those apps are not being adversely styled
